### PR TITLE
Fix out of bounds access in SetOrbitThreadsInTarget

### PR DESCRIPTION
**Pull request by beckerhe**
`SetOrbitThreadsInTarget` calls `SetOrbitThreads` in our user space instrumentation library through ptrace. To keep things simple `SetOrbitThreads` takes a fixed number of parameters which represent the TIDs of the injected threads (mostly from gRPC) to filter out events from them.

The TIDs of the injected threads are determined by comparing a list of TIDs from after the injection to a list from before the injection.

The code assumes that the resulting list contains at least 6 TIDs. But that recently changed. An update to the gRPC library reduced the number of threads and this now leads to an out of bounds access in the hard coded list.

This PR solves that by checking the size of the given list and providing a fall back value of -1 if the number of threads given in the list is shorter than 6.

Note that this still doesn't solve the case when we injected more than 6 threads. To solve this we probably have to change the data structure in the instrumentation library (from an array to a vector or similar) and I am not sure if this affects the performance in any way. That's why this is only fixing the OOB access.

### CODE REVIEW

File: src/UserSpaceInstrumentation/InstrumentProcess.cpp [hunk 1]
```diff
@@ -246,9 +247,13 @@ ErrorMessageOr<void> SetOrbitThreadsInTarget(pid_t pid, const std::vector<Module
...
   OUTCOME_TRY(void* set_orbit_threads_function_address,
               DlsymInTracee(pid, modules, library_handle, kSetOrbitThreadsFunctionName));
-  OUTCOME_TRY(ExecuteInProcess(pid, set_orbit_threads_function_address, orbit_threads[0],
-                               orbit_threads[1], orbit_threads[2], orbit_threads[3],
-                               orbit_threads[4], orbit_threads[5]));
+  const auto safe_get_tid = [&orbit_threads](size_t index) {
+    if (index < orbit_threads.size()) return orbit_threads.at(index);
+    return -1;
+  };
+  OUTCOME_TRY(ExecuteInProcess(pid, set_orbit_threads_function_address, safe_get_tid(0),
```
Comment (dpallotti) [inline src/UserSpaceInstrumentation/InstrumentProcess.cpp:254]: Hmm, maybe this should be better discussed with @danielfenner , but above we have `ORBIT_CHECK(orbit_threads.size() == GetExpectedOrbitThreadNames().size())`, so here we are already guaranteed to have exactly `orbit_threads.size() == 5`. It seems to me that we should just change `OrbitUserSpaceInstrumentation.cpp/SetOrbitThreads` to take five parameters (and the underlying array to have size 5), and that this was simply forgotten when updating gRPC.

In other words, making `OrbitUserSpaceInstrumentation.cpp` expect a fixed number of thread names may sound like something we want to fix, but in reality the whole injection code here relies on that number to be fixed.

Comment (dpallotti) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp:254]: Basically, 

> The code assumes that the resulting list contains at least 6 TIDs. But that recently changed.

is not the case. The code assumes a fixed number.

Comment (beckerhe) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp:254]: Oh yeah. You're right. This is more complicated than I thought. I'm gonna ignore this problem for now. Thanks for the review though!

Comment (dpallotti) [reply src/UserSpaceInstrumentation/InstrumentProcess.cpp:254]: But what about just changing `OrbitUserSpaceInstrumentation.cpp/SetOrbitThreads` to take five TIDs instead of six? If the number changes again this problem could appear again, but at least it should fix the  out-of-bounds access for now.
